data_handler/handlers/loan_states/hashtack_v0/interest_rate.py

```diff
@@ -0,0 +1,87 @@
+from decimal import Decimal
+
+from db.models import InterestRate
+from handler_tools.api_connector import DeRiskAPIConnector
+from handlers.loan_states.hashtack_v0.events import HashstackV0Event
+
+HASHSTACK_INTEREST_RATE_ADDRESS = "0x01b862c518939339b950d0d21a3d4cc8ead102d6270850ac8544636e558fab68"
+
+SECONDS_IN_YEAR = Decimal(365 * 24 * 60 * 60)
+
+
+class HashstackV0InterestRate:
+    """Class for calculating interest rates for a given token on the Hashstack V0 protocol."""
+
+    def __init__(self, token: str, min_block_number: int, max_block_number: int):
```

Comment:
I think it's more efficient to process events for all tokens at once (since we have to load all necessary data anyway). The `token` parameter will then become redundant.